--- unzip60/unix/Makefile	2009-01-19 05:41:18.000000000 +0700
+++ unzip60/unix/Makefile.new	2026-01-01 06:29:22.301664897 +0700
@@ -122,8 +122,8 @@
 # on some systems, manext=l and MANDIR=/usr/man/man$(manext) may be appropriate
 manext = 1
 prefix = /usr/local
-BINDIR = $(prefix)/bin#			where to install executables
-MANDIR = $(prefix)/man/man$(manext)#	where to install man pages
+BINDIR = ${DESTDIR}/$(prefix)/bin#			where to install executables
+MANDIR = ${DESTDIR}/$(prefix)/man/man$(manext)#	where to install man pages
 INSTALLEDBIN = $(BINDIR)/funzip$E $(BINDIR)/unzip$E $(BINDIR)/unzipsfx$E \
 	$(BINDIR)/zipgrep$E $(BINDIR)/zipinfo$E
 INSTALLEDMAN = $(MANDIR)/funzip.$(manext) $(MANDIR)/unzip.$(manext) \
