--- hyprtoolkit-0.5.3/CMakeLists.txt.orig	2026-02-12 09:33:14.877632229 +0700
+++ hyprtoolkit-0.5.3/CMakeLists.txt	2026-02-12 09:33:52.962630531 +0700
@@ -73,7 +73,7 @@
   PRIVATE "./src" "./src/include" "./protocols" "${CMAKE_BINARY_DIR}")
 set_target_properties(hyprtoolkit PROPERTIES VERSION ${HYPRTOOLKIT_VERSION}
                                              SOVERSION 5)
-target_link_libraries(hyprtoolkit PUBLIC OpenGL::EGL OpenGL::OpenGL
+target_link_libraries(hyprtoolkit PUBLIC OpenGL::EGL OpenGL::GL
                                          PkgConfig::deps)
 
 check_include_file("sys/timerfd.h" HAS_TIMERFD)
@@ -172,7 +172,7 @@
     PRIVATE "./src" "./src/include" "./protocols" "${CMAKE_BINARY_DIR}")
   target_link_libraries(
     hyprtoolkit_tests PRIVATE hyprtoolkit GTest::gtest_main OpenGL::EGL
-                                     OpenGL::OpenGL PkgConfig::deps)
+                                     OpenGL::GL PkgConfig::deps)
 
   gtest_discover_tests(hyprtoolkit_tests)
 
