--- shadow-4.20.0/lib/find_new_sub_uids.c	2026-07-21 04:15:36.428675754 +0700
+++ shadow-4.20.0/lib/find_new_sub_uids.c.new	2026-07-31 09:14:47.491453577 +0700
@@ -8,6 +8,7 @@
 
 #ifdef ENABLE_SUBIDS
 
+#include <stdint.h>
 #include <stdio.h>
 #include <errno.h>
 
--- shadow-4.20.0/lib/find_new_sub_gids.c	2026-07-21 04:15:36.428675754 +0700
+++ shadow-4.20.0/lib/find_new_sub_gids.c.new	2026-07-31 09:14:36.091611571 +0700
@@ -8,6 +8,7 @@
 
 #ifdef ENABLE_SUBIDS
 
+#include <stdint.h>
 #include <stdio.h>
 #include <errno.h>
 
